hw/ip/otbn/dv/otbnsim/sim/insn.py

```diff
@@ -16,20 +16,22 @@
 class ADD(RV32RegReg):
     insn = insn_for_mnemonic('add', 3)
 
-    def execute(self, state: OTBNState) -> None:
+    def execute(self, state: OTBNState) -> Optional[Iterator[None]]:
         val1 = state.gprs.get_reg(self.grs1).read_unsigned()
         val2 = state.gprs.get_reg(self.grs2).read_unsigned()
         result = (val1 + val2) & ((1 << 32) - 1)
         state.gprs.get_reg(self.grd).write_unsigned(result)
+        return None
```

Comment:
You don't need to explicitly return None, but you can, of course (mypy passed on my prototype commit when the non-generator functions had still `None` as return type, instead of `Optional[Iterator[None]]`. No strong feelings either way.

Reply:
Ah! That's neat! I think I'd probably prefer to keep the types uniform and remove the `return None` as a documentation rendering step. Otherwise, you still get the return lines at the end of the multi-cycle instructions, which is a bit ugly.

Reply:
That's neat! Unfortunately, we'd still have to keep the `return None` lines at the end of multi-cycle instructions. Maybe it's better to do it as a documentation post-processing step.

Reply:
Really? I didn't have it and it worked just fine for mypy. And IIRC, "return None" and "return" and no "return" at all are functionally equivalent in Python. But as I said, really up to you.

Reply:
Ahah! You're right! It only moans on the non-generator functions! New version coming right up...